### PR TITLE
Implementation of the search assets in firebase

### PR DESCRIPTION
**Pull request by ezequielsetzesglobant**
The feature of bringing assets from Firebase with the filter functionality was implemented as well.
In addition, internet validations were made.

Videos case filter by barcode and case all:
https://user-images.githubusercontent.com/84338513/152569705-c8f4d1f8-0698-4778-9e9a-8b14c8ae479e.mov
https://user-images.githubusercontent.com/84338513/152569807-c2c5baf9-fbf8-431d-b3e8-1fbe9a7e7da2.mov
https://user-images.githubusercontent.com/84338513/152569895-ee7f71be-ae4d-4f5c-9d11-0b8b59451c46.mov

Videos case internet validation in camera screen and assets list screen:
https://user-images.githubusercontent.com/84338513/152570025-c416ff11-2a43-4332-bd3f-7d77e6867282.mov
https://user-images.githubusercontent.com/84338513/152570108-648ad0b2-2228-4034-aea7-3a4d0b06ebe3.mov
https://user-images.githubusercontent.com/84338513/152570196-be91d1de-996b-49f1-8a88-e73290ab998d.mov

Unit test passed:
Unit test passed 1
Unit test passed 2

Coverage:
Coverage

### CODE REVIEW

File: stockApp/stockApp/Presentation/View/AssetViewController.swift [hunk 1]
```diff
@@ -21,6 +21,8 @@ class AssetViewController: UIViewController {
 
 	var assets: [AssetViewData] = []
 
+    let reachability = try? Reachability()
+
 	var helper: HelperFunctions
 
 	let searchAlertView = SearchAlertView()
```
Comment (ezequielvelezglobant) [inline stockApp/stockApp/Presentation/View/AssetViewController.swift:28]: fix alignment of all document

File: stockApp/stockApp/Presentation/View/CameraViewController.swift [hunk 2]
```diff
@@ -356,21 +358,26 @@ extension CameraViewController: UIPickerViewDelegate, UIPickerViewDataSource {
...
+            alertHelper.showAlertMessage(title: Strings.isNotConnectedToInternet, viewController: self, handler: {_ in
+                self.searchAlertView.dismissAlert()
+            })
+        } else {
+            searchAlertView.dismissAlert()
+            dismiss(animated: true, completion: {
+                self.coordinator?.goToAssetViewController(filter: filter, value: value)
+            })
+        }
+        self.reachability?.stopNotifier()
```
Comment (ezequielvelezglobant) [inline stockApp/stockApp/Presentation/View/CameraViewController.swift:377]: fix alignment

File: stockApp/stockApp/Presentation/View/CameraViewController.swift [hunk 3]
```diff
@@ -356,21 +358,26 @@ extension CameraViewController: UIPickerViewDelegate, UIPickerViewDataSource {
...
 	}
 
 	func showErrorMesage(error: String) {
-		alertHelper.showAlertMessage(title: "Error", message: error, viewController: self)
+        alertHelper.showAlertMessage(title: "Error", message: error, viewController: self, handler: {_ in })
```
Comment (ezequielvelezglobant) [inline stockApp/stockApp/Presentation/View/CameraViewController.swift:381]: fix alignment

File: stockApp/stockAppTests/AssetViewModelTest.swift [hunk 4]
```diff
@@ -11,9 +11,9 @@ class MockAssetView {
 
     var assets: [AssetViewData] = []
 
-    var displayAssetsValue: Bool = false
-    var displayNotAssetsFoundMessageValue: Bool = false
-    var displayThereIsNotInternetMessageValue: Bool = false
+	var displayAssetsValue: Bool = false
+	var displayNotAssetsFoundMessageValue: Bool = false
+	var displayRequestErrorMessageValue: Bool = false
```
Comment (ezequielvelezglobant) [inline stockApp/stockAppTests/AssetViewModelTest.swift:16]: fix alignment.
review that maybe you have a diferent alignment configuration

**Review by ezequielvelezglobant — CHANGES_REQUESTED** [review]

File: stockApp/stockAppTests/AssetViewModelTest.swift (hunk 4 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockAppTests/AssetViewModelTest.swift]: done

File: stockApp/stockApp/Presentation/View/CameraViewController.swift (hunk 3 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Presentation/View/CameraViewController.swift:381]: done

File: stockApp/stockApp/Presentation/View/CameraViewController.swift (hunk 2 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Presentation/View/CameraViewController.swift:377]: done

File: stockApp/stockApp/Presentation/View/AssetViewController.swift (hunk 1 above)
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Presentation/View/AssetViewController.swift]: done

File: stockApp/stockApp/Domain/UseCase/GetAssetUseCase.swift [hunk 5]
```diff
@@ -12,18 +12,57 @@ class GetAssetUseCase: GetAssetUseCaseProtocol {
...
+        case FilterType.assetName.rawValue:
+            return asset.assetName.lowercased().contains(searchText.lowercased())
+        case FilterType.userMail.rawValue:
+            return asset.userMail.lowercased().contains(searchText.lowercased())
+        case FilterType.location.rawValue:
+            return asset.location.lowercased().contains(searchText.lowercased())
+        default:
+            return false
+        }
+    }
```
[the remark this answers is not in the document]
Comment (ezequielsetzesglobant) [reply stockApp/stockApp/Domain/UseCase/GetAssetUseCase.swift]: done

**Review by Faka-GLB — APPROVED** [review]

**Review by ezequielvelezglobant — APPROVED** [review]

**Pull request merged**